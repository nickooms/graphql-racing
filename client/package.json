{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "apollo-client": "^1.1.1",
    "apollo-test-utils": "^0.1.1",
    "false": "0.0.4",
    "graphql": "^0.10.1",
    "graphql-tools": "^0.10.0",
    "react": "^15.4.2",
    "react-apollo": "^1.1.2",
    "react-dom": "^15.4.2",
    "react-router-dom": "^4.0.0",
    "subscriptions-transport-ws": "^0.7.3",
    "testcafe": "^0.17.0",
    "testcafe-react-selectors": "0.0.6",
    "ws": "^3.1.0"
  },
  "devDependencies": {
    "babel-jest": "^20.0.3",
    "babel-preset-es2015": "^6.24.1",
    "babel-preset-react": "^6.24.1",
    "bufferutil": "^3.0.2",
    "jest": "^20.0.4",
    "react-scripts": "0.9.3",
    "react-test-renderer": "^15.6.1",
    "utf-8-validate": "^3.0.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest --coverage",
    "eject": "react-scripts eject",
    "testcafe": "testcafe chrome testcafe/tests/*"
  }
}
