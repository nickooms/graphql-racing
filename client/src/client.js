import { ApolloClient, createNetworkInterface, toIdValue } from 'react-apollo';
import { SubscriptionClient, addGraphQLSubscriptions } from 'subscriptions-transport-ws';
// import WebSocket from 'ws';

const networkInterface = createNetworkInterface({ uri: 'http://localhost:4000/graphql' });
networkInterface.use([{
  applyMiddleware(req, next) {
    setTimeout(next, 500);
  },
}]);

const wsClient = new SubscriptionClient(`ws://localhost:4000/subscriptions`, {
  reconnect: true
}/* , WebSocket*/);

const networkInterfaceWithSubscriptions = addGraphQLSubscriptions(
  networkInterface,
  wsClient
);

function dataIdFromObject (result) {
  if (result.__typename) {
    if (result.id !== undefined) {
      return `${result.__typename}:${result.id}`;
    }
  }
  return null;
}

const client = new ApolloClient({
  networkInterface: networkInterfaceWithSubscriptions,
  customResolvers: {
    Query: {
      channel: (_, args) => {
        return toIdValue(dataIdFromObject({ __typename: 'Channel', id: args['id'] }))
      },
    },
  },
  dataIdFromObject,
});

export default client;
