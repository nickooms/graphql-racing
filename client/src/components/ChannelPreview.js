import { graphql } from 'react-apollo';

import query from '../query/channel';

import ChannelPreview from './functional-stateless/ChannelPreview';

export default (graphql(query, {
  options: (props) => ({
    variables: { channelId: props.channelId },
  }),
})(ChannelPreview));
