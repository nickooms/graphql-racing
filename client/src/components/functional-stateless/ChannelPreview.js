import React from 'react';

const ChannelPreview = ({ data: { loading, error, channel } }) => {
  if (loading) return (
    <p>Loading ...</p>
  );
  if (error) return (
    <p>{error.message}</p>
  );
  const { name } = channel;
  return (
    <div>
      <div className="channelName">{name}</div>
      <div>Loading Messages</div>
    </div>
  );
};

export default ChannelPreview;
