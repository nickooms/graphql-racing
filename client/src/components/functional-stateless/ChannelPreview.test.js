import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';
import React from 'react';

import ChannelPreview from './ChannelPreview';
import ChannelData from '../__testdata__/Channel';
import client from '../../test-client';

const { data } = ChannelData;

test('ChannelPreview renders appropriately', () => {
  const component = renderer.create(
    <ApolloProvider client={client}>
      <ChannelPreview data={data} />
    </ApolloProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ChannelPreview error renders appropriately', () => {
  const component = renderer.create(
    <ApolloProvider client={client}>
      <ChannelPreview data={{ error: { message: 'ERROR' } }} />
    </ApolloProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});