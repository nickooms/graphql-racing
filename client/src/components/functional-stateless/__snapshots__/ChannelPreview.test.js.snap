// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPreview error renders appropriately 1`] = `
<p>
  ERROR
</p>
`;

exports[`ChannelPreview renders appropriately 1`] = `
<div>
  <div
    className="channelName"
  >
    soccer
  </div>
  <div>
    Loading Messages
  </div>
</div>
`;
